src/main/java/com/yahoo/bullet/storm/FilterBolt.java

```diff
@@ -91,50 +80,77 @@ public void execute(Tuple tuple) {
 
     @Override
     public void declareOutputFields(OutputFieldsDeclarer declarer) {
-        declarer.declare(new Fields(TopologyConstants.ID_FIELD, TopologyConstants.RECORD_FIELD));
+        // This is the per query data stream
+        declarer.declareStream(DATA_STREAM, new Fields(ID_FIELD, DATA_FIELD));
+        // This is the where any errors per query are sent
+        declarer.declareStream(ERROR_STREAM, new Fields(ID_FIELD, ERROR_FIELD));
     }
 
-    @Override
-    protected FilterQuery createQuery(Tuple queryTuple) {
-        String queryString = queryTuple.getString(TopologyConstants.QUERY_POSITION);
-        // No need to handle any errors here. The JoinBolt reports all errors.
+    private void onQuery(Tuple tuple) {
+        String id = tuple.getString(TopologyConstants.ID_POSITION);
+        String query = tuple.getString(TopologyConstants.QUERY_POSITION);
+
+        // No need to handle any errors in the Filter Bolt.
+        Querier querier = null;
         try {
-            return new FilterQuery(queryString, configuration);
-        } catch (ParsingException | RuntimeException e) {
-            return null;
+            querier = createQuerier(id, query, config);
```

Comment:
I don't need to set querier to null when having errors, just log it and return.

Reply:
I set it to null to print the log.error. There are two ways to fail. RuntimeException or initialize() has errors.

Reply:
Yup, my bad. Fixed and simplified.

Reply:
You can even merge line 94 to line 95.